demos/device_defender_for_aws/metrics_collector.h

```diff
@@ -0,0 +1,140 @@
+/*
+ * FreeRTOS V202011.00
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * http://aws.amazon.com/freertos
+ * http://www.FreeRTOS.org
+ */
+
+#ifndef __METRICS_COLLECTOR_H__
+#define __METRICS_COLLECTOR_H__
+
+/* Standard includes. */
+#include <stdint.h>
+
+/**
+ * @brief Return codes from metrics collector APIs.
+ */
+typedef enum
+{
+    MetricsCollectorSuccess = 0,
+    MetricsCollectorBadParameter,
+    MetricsCollectorCollectionFailed
+} MetricsCollectorStatus_t;
+
+/**
+ * @brief Represents network stats.
+ */
+typedef struct NetworkStats
+{
+    uint32_t bytesReceived;   /* Number of bytes received. */
+    uint32_t bytesSent;       /* Number of bytes sent. */
+    uint32_t packetsReceived; /* Number of TCP packets received. */
+    uint32_t packetsSent;     /* Number of TCP packets sent. */
+} NetworkStats_t;
+
+/**
+ * @brief Represents a network connection.
+ */
+typedef struct Connection
+{
+    uint32_t localIp;
+    uint32_t remoteIp;
+    uint16_t localPort;
+    uint16_t remotePort;
+} Connection_t;
+
+/**
+ * @brief Get network stats.
+ *
+ * @param[out] pOutNetworkStats The network stats.
+ *
+ * @return #MetricsCollectorSuccess if the network stats are successfully obtained;
+ * #MetricsCollectorBadParameter if invalid parameters are passe;
+ * #MetricsCollectorCollectionFailed if the collection methods failed.
+ */
+MetricsCollectorStatus_t GetNetworkStats( NetworkStats_t * pOutNetworkStats );
+
+/**
+ * @brief Get a list of the open TCP ports.
+ *
+ * This function can be called with @p pOutTcpPortsArray NULL to get the number
```

Comment:
Is this required behavior for this API or an artifact of the way it is implemented?  If it is the latter I would delete this sentence.

Reply:
It is meant to be required behavior for the implementation.